### ISSUE TITLE
Problems at the language server and linting

### ISSUE THREAD

**Issue by rhuanbarreto** [rubyide/vscode-ruby#500]
### Your environment

- `vscode-ruby` version: 0.23.0
- Ruby version: ruby 2.6.2p47 (2019-03-13 revision 67232) [x86_64-darwin18]
- Ruby version manager (if any): rbenv
- VS Code version: 1.35.1
- Operating System: MacOS X 10.14.5
- Using language server? (eg `useLanguageServer` is true?) Yes
- Rubocop version: 0.72.0 (using Parser 2.6.2.1, running on ruby 2.6.2 x86_64-darwin18)

### Actual behavior

Linting is not working even though I enabled in the config:

```
    "ruby.format": "rubocop",
    "ruby.intellisense": "rubyLocate",
    "ruby.codeCompletion": "rcodetools",
    "ruby.lint": {
        "rubocop": {
            "lint": true,
            "rails": true
        }
    },
    "[ruby]": {
        "editor.formatOnSave": true,
        "editor.defaultFormatter": "rebornix.ruby"
    },
    "ruby.useLanguageServer": true,
```

Also in the output pane I get this error:

```
[2019-06-27 09:32:04.851] [renderer1] [error] Cannot read property 'then' of undefined: TypeError: Cannot read property 'then' of undefined
	at Locate.listInFile (/Users/rhuan/.vscode/extensions/rebornix.ruby-0.23.0/out/locate/locate.js:69:29)
	at Object.provideDocumentSymbols (/Users/rhuan/.vscode/extensions/rebornix.ruby-0.23.0/out/providers/intellisense.js:55:31)
	at define.provideDocumentSymbols.a.asPromise (/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/services/extensions/node/extensionHostProcess.js:572:267)
	at t.asPromise.Promise (/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/services/extensions/node/extensionHostProcess.js:53:320)
	at new Promise (<anonymous>)
	at Object.t.asPromise (/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/services/extensions/node/extensionHostProcess.js:53:292)
	at _.provideDocumentSymbols (/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/services/extensions/node/extensionHostProcess.js:572:238)
	at define.$provideDocumentSymbols._withAdapter.e (/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/services/extensions/node/extensionHostProcess.js:588:888)
	at j._withAdapter (/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/services/extensions/node/extensionHostProcess.js:588:126)
	at j.$provideDocumentSymbols (/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/services/extensions/node/extensionHostProcess.js:588:866)
	at d._doInvokeHandler (/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/services/extensions/node/extensionHostProcess.js:644:1006)
	at d._invokeHandler (/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/services/extensions/node/extensionHostProcess.js:644:698)
	at d._receiveRequest (/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/services/extensions/node/extensionHostProcess.js:643:293)
	at d._receiveOneMessage (/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/services/extensions/node/extensionHostProcess.js:642:141)
	at define.constructor._protocol.onMessage.e (/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/services/extensions/node/extensionHostProcess.js:640:400)
	at u.fire (/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/services/extensions/node/extensionHostProcess.js:47:720)
	at a (/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/services/extensions/node/extensionHostProcess.js:176:766)
	at e (/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/services/extensions/node/extensionHostProcess.js:176:812)
	at u.fire (/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/services/extensions/node/extensionHostProcess.js:47:720)
	at n.constructor.e.onMessage.e (/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/services/extensions/node/extensionHostProcess.js:748:721)
	at u.fire (/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/services/extensions/node/extensionHostProcess.js:47:720)
	at a (/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/services/extensions/node/extensionHostProcess.js:176:766)
	at e (/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/services/extensions/node/extensionHostProcess.js:176:812)
	at u.fire (/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/services/extensions/node/extensionHostProcess.js:47:720)
	at y._receiveMessage (/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/services/extensions/node/extensionHostProcess.js:185:670)
	at define.constructor._socketDisposables.push._socketReader.onMessage.e (/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/services/extensions/node/extensionHostProcess.js:182:981)
	at u.fire (/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/services/extensions/node/extensionHostProcess.js:47:720)
	at f.acceptChunk (/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/services/extensions/node/extensionHostProcess.js:179:532)
	at define.constructor._register._socket.onData.e (/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/services/extensions/node/extensionHostProcess.js:178:869)
	at Socket.t (/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/services/extensions/node/extensionHostProcess.js:187:141)
	at Socket.emit (events.js:182:13)
	at addChunk (_stream_readable.js:279:12)
	at readableAddChunk (_stream_readable.js:264:11)
	at Socket.Readable.push (_stream_readable.js:219:10)
	at Pipe.onread (net.js:636:20)
```

Also on the output of the language server:

```
[Info  - 9:27:34 AM] Initializing Ruby language server...
[Info  - 9:27:34 AM] Rebuilding tree-sitter for local Electron version
[Info  - 9:27:34 AM] Rebuild succeeded!
Lint: executing rubocop -s /**************file.rb -f json -R -l...
invalid option: -R

For usage information, use --help

Lint: Received invalid JSON from rubocop:
```

### Expected behavior

Linting occur normally.

**Comment by wingrunr21** [comment]
Rails cops were extracted to a separate gem in 0.72.0:

https://github.com/rubocop-hq/rubocop/issues/5976

**Comment by rhuanbarreto** [comment]
After I removed the rails linting: 

```
    "ruby.format": "rubocop",
    "ruby.intellisense": "rubyLocate",
    "ruby.codeCompletion": "rcodetools",
    "ruby.lint": {
        "rubocop": true
    },
    "[ruby]": {
        "editor.formatOnSave": true,
        "editor.defaultFormatter": "rebornix.ruby"
    },
    "ruby.useLanguageServer": true,
```

- Error in the extension host log still persist `[renderer1] [error] Cannot read property 'then' of undefined: TypeError: Cannot read property 'then' of undefined
	at Locate.listInFile (/Users/rhuan/.vscode/extensions/rebornix.ruby-0.23.0/out/locate/locate.js:69:29)`
- No problems are reported in the problems tab for linting. So no linting yet
- Linting command on the language server give no errors anymore.

**Comment by wingrunr21** [comment]
* `then` was fixed in #499 
* You need to look at the output and see what is being run and returned. I can't magically see what's happening on your machine.

**Comment by rhuanbarreto** [comment]
After a vscode restart it's working.
